### PR TITLE
Add circleCI flow #1

### PR DESCRIPTION
**Pull request by flamerecca**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+jobs: # basic units of work in a run
+  build: # runs not using Workflows must have a `build` job as entry point
+    docker: # run the steps with Docker
+      # CircleCI Go images available at: https://hub.docker.com/r/circleci/golang/
+      - image: circleci/golang:1.12 #
+    # directory where steps are run. Path must conform to the Go Workspace requirements
+    working_directory: /go/src/github.com/u-job/api-server
+
+    steps: # steps that comprise the `build` job
+      - checkout # check out source code to working directory
```
Comment (shana0440) [inline .circleci/config.yml]: please check following step
- go fmt
- go build

and please create a script for this.
you should add a scripts folder and put the ci script in that.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+      - restore_cache: # restores saved cache if no changes are detected since last run
+          # Read about caching dependencies: https://circleci.com/docs/2.0/caching/
+          keys:
+            - go-mod-{{ checksum "go.mod" }}
```
Comment (shana0440) [inline .circleci/config.yml]: you should restore cache before run steps. and download the dependencies after restore cache

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+      - save_cache: # Store cache in the /go/pkg directory
+          key: go-mod-{{ checksum "go.mod" }}
+          paths:
+            - "/go/pkg/mod"
```
Comment (shana0440) [inline .circleci/config.yml]: are you sure the package is store in this path?

Comment (flamerecca) [reply .circleci/config.yml]: 這是參考範例專案 github 的做法
https://github.com/CircleCI-Public/circleci-demo-go/blob/master/.circleci/config.yml#L47

Comment (shana0440) [reply .circleci/config.yml]: I see, thanks.

**Comment by shana0440** [comment]
since I will take over, so close it.

**Pull request closed without merging**